File: android/src/main/kotlin/com/giphy/giphy_flutter_sdk/GiphyFlutterMediaViewFactory.kt
package com.giphy.giphy_flutter_sdk

import com.giphy.giphy_flutter_sdk.GiphyFlutterMediaView
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import io.flutter.plugin.common.StandardMessageCodec
import android.content.Context
import io.flutter.plugin.common.BinaryMessenger

class GiphyFlutterMediaViewFactory(private val messenger: BinaryMessenger) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as Map<String, Any>?
        return GiphyFlutterMediaView(context, messenger, viewId, creationParams)
    }
}

File: ios/giphy_flutter_sdk/Sources/giphy_flutter_sdk/GiphyFlutterMediaViewFactory.swift
//
//  GiphyFlutterMediaViewFactory.swift
//  giphy_flutter_sdk
//
//  Created by Alex on 27.2.2024.
//

import Flutter
import UIKit

public class GiphyFlutterMediaViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
    
    public init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }
    
    public func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        return GiphyFlutterMediaView(frame: frame, viewId: viewId, args: args, binaryMessenger: messenger)
    }
}
